f5_openstack_agent/lbaasv2/drivers/bigip/selfips.py

```diff
@@ -180,6 +180,16 @@ def _get_bigip_selfip_address(self, bigip, subnet, device_id):
                 host_passed=host_passed
             )
 
+            if port:
+                ports = self.driver.plugin_rpc.get_port_by_name(
+                    port_name=selfip_name)
+                # If the port created by me is not the first, that means
+                # another agent process creates a new port before me.
+                # Delete my port and use the first one.
+                if len(ports) > 0 and ports[0]['id'] != port['id']:
```

Comment:
Shall we use `try .. except .. ` to catch port delete error, because there may happen that 2 agents concurrently found extra ports for deletion.

Reply:
> Shall we use `try .. except .. ` to catch port delete error, because there may happen that 2 agents do extra port deletions concurrently.

The first port got by those agent processes are identical, and each agent process will delete the port create by itself. There will not be any conflict.